File: frontend/src/api/notification.ts
import api from './client'

export interface NotificationItem {
  id: number
  user_id: number
  type: string
  title: string
  message: string
  reference_type: string | null
  reference_id: number | null
  is_read: boolean
  created_at: string
}

export interface PaginatedNotifications {
  items: NotificationItem[]
  total: number
  page: number
  page_size: number
}

export interface UnreadCount {
  count: number
}

export const notificationApi = {
  list: (params?: { page?: number; page_size?: number; unread_only?: boolean }) =>
    api.get<PaginatedNotifications>('/notifications', { params }),

  unreadCount: () =>
    api.get<UnreadCount>('/notifications/unread-count'),

  markRead: (id: number) =>
    api.post<NotificationItem>(`/notifications/${id}/read`),

  markAllRead: () =>
    api.post<UnreadCount>('/notifications/read-all'),
}

File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    type: str
    title: str
    message: str
    reference_type: str | None = None
    reference_id: int | None = None
    is_read: bool
    created_at: datetime


class PaginatedNotifications(BaseModel):
    items: list[NotificationOut]
    total: int
    page: int
    page_size: int


class UnreadCountOut(BaseModel):
    count: int
